gitregostore/gitstoremethods.go

```diff
@@ -106,16 +108,54 @@ func (gs *GitRegoStore) GetOPAControlByID(controlID string) (*opapolicy.Control,
 		if strings.EqualFold(control.ControlID, controlID) ||
 			// If backward compatibility is supported,try to find if the controlID sent has a new controlID
 			(supportBackwardCompatibility && strings.EqualFold(control.ControlID, newControlID(controlID))) {
-			err := gs.fillRulesAndRulesIDsInControl(&control)
-			if err != nil {
-				return nil, err
+
+			if len(control.Rules) == 0 {
+				err := gs.fillRulesAndRulesIDsInControl(&control)
+				if err != nil {
+					return nil, err
+				}
 			}
+
 			return &control, nil
 		}
 	}
 	return nil, fmt.Errorf("control '%s' not found", controlID)
 }
 
+// getFrameworkControlByControlName - get opapolicy.Framework and control name and return the relevant control object
+
+func (gs *GitRegoStore) getFrameworkControlByControlName(fw *opapolicy.Framework, controlName string) (*opapolicy.Control, error) {
```

Comment:
func name is confusing, shouldn't it be getOPAControlByFrameworkAndControlName?
also - why do we need this function? Doesn't the one below answer our needs?

Reply:
fixed.